### DISCUSSION TITLE
Elastalert mapping error (match_body)

### DISCUSSION THREAD

**Discussion by Usman0137** [jertel/elastalert2#1192] [Q&A] [answered]
"Hello, I am encountering an issue while creating dashboards in Kibana using Wazuh SIEM and Elastalert. I have configured the custom rules with Elastalert, but when attempting to create dashboards in Kibana, I am facing a 'match_body' issue. I have attached a screenshot for reference.

**Comment by jertel** [comment]
What specifically is the issue?

**Reply by Usman0137** [reply]
1. I have added custom fields in my Elastalert rule file, but the fields are not getting mapped in Kibana. I want only those custom fields to be correctly mapped and visible in Kibana.
Image # 1 is Elastalert file screenshot

2. I have noticed that the same rule in Kibana shows different fields at different times. Sometimes it only shows a few fields, while other times it shows more fields. I want only the specific custom fields I added in the Elastalert file to be consistently shown in Kibana. I want to use these fields to create dashboards.
Image # 2 and Image # 3 is of the same rule in Kibana shows different fields at different time.

3. Lastly, the custom fields are not being cached in Elasticsearch even after refreshing the elastalert* index as suggested by the error message tag and even after deleting and recreating it but still the issue persists. Because of this the custom fields that we are trying to add cannot be used in creating visualizations.  
Image # 4 reference image.

Image # 1 
![image](https://github.com/jertel/elastalert2/assets/18024022/75717859-cde5-4772-bf40-5094d23d28bf)

**Reply by jertel** [reply]
1 - The `elastalert_*` indices are internal to ElastAlert 2 and not intended for any other purpose. They are used for tracking which rules have run recently, which data to use for subsequent runs when calculating matches, how long to silence future alerts, etc. There is no predefined schema and the schema can change between releases. Also there have been past releases that require the deletion of these indices.

2 - You are comparing two different indices. `elastalert_status` vs `elastalert_status_status`. They will never have the same field mappings as the two indices are used for different purposes.

3 - Custom fields are intended to be used for populating alert contents, such as email bodies or slack messages. 

I suggest using the original indices for visualizations, e.g., the same indices that you are specifying in your rule yaml files, such as `wazuh-archives-*`

**Reply by Usman0137** [reply]
No, I think I can't explain it clearly so let's make it simple I only want to display the fields listed under the "include" tag in the rule file of Elastalert in Kibana.

Image # 1 The fields I added under the include tag.

![image](https://github.com/jertel/elastalert2/assets/18024022/96aef306-1d0b-44b2-a358-b09f8fe4c289)

Image # 2 only wants to map those fields here in Kibana not all the payloads like this.

![1111](https://github.com/jertel/elastalert2/assets/18024022/711e4ee2-d548-4331-b2e6-22f6b3ba2f24)

**Reply by jertel** [reply]
Following your keep-it-simple approach, let me ask this question:

Do you agree that the screenshots you just posted above are showing data records from internal `elastalert_*` indices?

**Reply by Usman0137** [reply]
Yes, the above posted screenshots are from elastalert_* index

**Reply by jertel** [reply]
Ok. Why are you showing me a screenshot of data inside that internal index?

**Reply by Usman0137** [reply]
Because the problem is that I don't want all the data to be included in the index. I only want the field I added to the ElastAlert rule file to be displayed in the index.

**Reply by jertel** [reply]
Why are you concerned with that index?

**Reply by Usman0137** [reply]
I am concerned about that index because I have customized rules in ElastAlert that are specifically designed for my use cases. When an alert is triggered, the corresponding data is stored in the ElastAlert_* index. My goal is to create dashboards using the ElastAlert index to visualize and analyze the data generated by these rule triggers. However, despite only including a few fields in the ElastAlert rule file, the resulting index contains numerous irrelevant fields that are of no use to me. so I want only that data in the Elastalert_* index which I added in the Elastalert rule file when it generated the alert.

**Reply by jertel** [reply]
As I mentioned in my reply 5 days ago, the contents of those _internal_ ElastAlert 2 indices are not intended to be used for dashboarding or any other purpose. They contain the contents of the document that triggered the alert. The `include:` block is not relevant to what's stored in that index. Per the documentation:

![image](https://github.com/jertel/elastalert2/assets/16639611/bd49cb6e-4009-463c-a98a-2857f54533d4)

This is stating that the `include:` fields adjust what data is sent in the alert message. It does not state that it alters what data is written to the internal-only ElastAlert 2 indices.

**Reply by Usman0137** [reply]
You mentioned that the alert includes the content of the document, which triggers the alert. I need assistance in resolving the cache mapping error so that I can create dashboards with visualizations for source IP, destination IP, and other related information. Please guide me on how to resolve this error (cache mapping).

**Reply by jertel** [reply]
Please paste the actual ElastAlert 2 error message with accompanying logs.

**Reply by Usman0137** [reply]
Below mentioned in the screenshot is the encountered error message and followed the recommended steps to fix it. I refreshed the index and even tried deleting and recreating it, but the issue still persists. I need assistance in resolving this cache mapping error.

![image](https://github.com/jertel/elastalert2/assets/18024022/98d75968-a188-492d-949a-d1f076c00cd1)

**Reply by jertel** [reply, marked as answer]
That's not an ElastAlert 2 error. Nor is that an error at all. It's a message stating that those fields are not mapped. They are not mapped because that is transient data only to be used internally to ElastAlert 2. 

We do not support using that index for visualizations.
